packages/firestore/src/lite-api/aggregate.ts

```diff
@@ -15,72 +15,114 @@
  * limitations under the License.
  */
 
-import { Value } from '../protos/firestore_proto_api';
-import { invokeRunAggregationQueryRpc } from '../remote/datastore';
 import { hardAssert } from '../util/assert';
 import { cast } from '../util/input_validation';
+import { invokeRunAggregationQueryRpc } from '../remote/datastore';
+import { Value } from '../protos/firestore_proto_api';
 
-import { getDatastore } from './components';
+import { DocumentFieldValue, Query, queryEqual } from './reference';
 import { Firestore } from './database';
-import { Query, queryEqual } from './reference';
+import { getDatastore } from './components';
 import { LiteUserDataWriter } from './reference_impl';
 
+import { deepEqual } from '@firebase/util';
+import { FieldPath } from './field_path';
+
 /**
- * An `AggregateQuery` computes some aggregation statistics from the result set of
- * a base `Query`.
+ * An `AggregateField` computes some aggregation statistics from the result set of
+ * an aggregation query.
  */
-export class AggregateQuery {
-  readonly type = 'AggregateQuery';
-  /**
-   * The query on which you called `countQuery` in order to get this `AggregateQuery`.
-   */
-  readonly query: Query<unknown>;
+export class AggregateField<T> {
+  type = 'AggregateField';
+  _datum?: T;
 
-  /** @hideconstructor */
-  constructor(query: Query<unknown>) {
-    this.query = query;
-  }
+  constructor(readonly subType: string) {}
 }
 
 /**
- * An `AggregateQuerySnapshot` contains results of a `AggregateQuery`.
+ * The union of all `AggregateField` types that are returned from the factory
+ * functions.
  */
-export class AggregateQuerySnapshot {
+export type AggregateFieldType =
+  | AggregateField<number>
+  | AggregateField<DocumentFieldValue | undefined>
+  | AggregateField<number | undefined>;
+
+/**
+ * A type whose values are all `AggregateField` objects.
+ * This is used as an argument to the "getter" functions, and the snapshot will
+ * map the same names to the corresponding values.
+ */
+export type AggregateSpec = { [field: string]: AggregateFieldType };
+
+/**
+ * A type whose keys are taken from an `AggregateSpec` type, and whose values
+ * are the result of the aggregation performed by the corresponding
+ * `AggregateField` from the input `AggregateSpec`.
+ */
+export type AggregateSpecData<T extends AggregateSpec> = {
+  [Property in keyof T]-?: T[Property]['_datum'];
+};
+
+/**
+ * Creates and returns an aggregation field that counts the documents in the result set.
+ * @returns An `AggregateField` object that includes number of documents.
+ */
+export function count(): AggregateField<number> {
+  return new AggregateField<number>('count');
+}
+
+/**
+ * Compares two `AggregateField` instances for equality.
+ * The two `AggregateField` instances are considered "equal" if and only if
+ * they were created by the same factory function (e.g. `count()`, `min()`, and
+ * `sum()`) with "equal" arguments.
+ */
+export function aggregateFieldEqual(
```

Comment:
use constructor type (both created by AggregateField class) and subType (both created by same function: min(),sum(),count()...) to check aggregateField equality